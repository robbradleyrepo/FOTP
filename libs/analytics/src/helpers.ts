export const gidToId = (gid: string) => {
  const parts = atob(gid).split("/");
  return parts[parts.length - 1];
};
