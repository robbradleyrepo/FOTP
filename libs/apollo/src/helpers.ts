import { defaultDataIdFromObject, IdGetterObj } from "apollo-cache-inmemory";
import {
  ApolloQueryResult,
  OperationVariables,
  QueryOptions,
} from "apollo-client";
import { ApolloError, isApolloError } from "apollo-client";
import type { FetchResult } from "apollo-link";
import { DocumentNode, GraphQLError } from "graphql";
import { ExecutionResult } from "react-apollo";

import { ApolloClientType } from "./types";

export { isApolloError };

export const dataIdFromObject = (
  obj: IdGetterObj & { [k: string]: any } // eslint-disable-line @typescript-eslint/no-explicit-any
) => {
  let dataId = defaultDataIdFromObject(obj);

  const { __typename, _meta } = obj;

  if (dataId === null && __typename) {
    // Get the ID from Prismic's `_meta` object, if available
    const ident = _meta?.id;

    if (ident) {
      dataId = `${__typename}:${ident}`;
    }
  }

  // If we're using a completed checkout, we'll add a suffix to ensure that
  // it's updated independently to the incomplete one
  if (__typename === "RCheckout" && obj.completedAt) {
    dataId += ":completed";
  }

  return dataId;
};

// TODO: improve types so `__typename` is explicitly excluded
export const sanitize = <T>(arg: T): T => {
  if (Array.isArray(arg)) {
    return (arg.map(sanitize) as unknown) as T;
  } else if (Object.prototype.toString.call(arg) === "[object Object]") {
    return (Object.entries(arg)
      .filter(([key]) => key !== "__typename")
      .reduce(
        (accum, [key, value]) => ({ ...accum, [key]: sanitize(value) }),
        {}
      ) as unknown) as T;
  }

  return arg;
};

const stringifyError = <T extends Error, K extends keyof T>(error: T) =>
  JSON.stringify(
    Object.getOwnPropertyNames(error).reduce(
      (accum, key) => ({ ...accum, [key]: JSON.stringify(error[key as K]) }),
      {}
    )
  );

export const stringifyApolloError = (error: ApolloError) =>
  stringifyError({
    ...error,
    graphQLErrors: error.graphQLErrors.map(stringifyError),
    networkError: error.networkError && stringifyError(error.networkError),
  });

export const throwGraphQLErrors = (
  results:
    | ApolloQueryResult<unknown>
    | ApolloQueryResult<unknown>[]
    | ExecutionResult<unknown>
    | ExecutionResult<unknown>[]
    | FetchResult<unknown>
    | FetchResult<unknown>[],
  errorMessage = "Unexpected GraphQLErrors"
) => {
  const allResults = Array.isArray(results) ? results : [results];
  const graphQLErrors: GraphQLError[] = allResults.reduce(
    (accum, { errors = [] }) => [...accum, ...errors],
    [] as GraphQLError[]
  );

  if (graphQLErrors.length > 0) {
    throw new ApolloError({
      errorMessage,
      extraInfo: {
        data: allResults.map(({ data }) => data),
      },
      graphQLErrors,
    });
  }
};

export const runServerSideQuery = async <
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  T = Record<string, any>,
  TVariables = OperationVariables
>(
  client: ApolloClientType,
  queryOrOptions: DocumentNode | QueryOptions<TVariables>,
  errorMessage = "Unexpected GraphQLErrors"
) => {
  const result = await client.query<T, TVariables>({
    fetchPolicy: "no-cache",
    ...("query" in queryOrOptions ? queryOrOptions : { query: queryOrOptions }),
  });

  throwGraphQLErrors(result, errorMessage);

  return result;
};
