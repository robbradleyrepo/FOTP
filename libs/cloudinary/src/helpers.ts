import CLOUDINARY, { CLOUDINARY_ORIGIN } from "./config";
import {
  CloudinaryFetchedImageSrcSetParams,
  CloudinaryFetchedImageUrlParams,
  CloudinaryMedia,
  CloudinaryUploadSrcSetParams,
  CloudinaryUploadUrlParams,
  CloudinaryUrlParams,
} from "./types";

const getParamsString = ({
  flags = [],
  format = "auto",
  quality = "auto",
  width = "auto",
}: CloudinaryUrlParams) =>
  [
    `f_${format}`,
    ...flags.map((flag) => `fl_${flag}`),
    `q_${quality}`,
    `w_${width}`,
  ]
    .sort() // Use consistent sorting for improved caching
    .join(",");

export const getFetchedImageUrl = (params: CloudinaryFetchedImageUrlParams) =>
  [
    CLOUDINARY_ORIGIN,
    params.cloudName ?? CLOUDINARY.defaultCloudName,
    params.type ?? CloudinaryMedia.IMAGE,
    "fetch",
    getParamsString(params),
    encodeURIComponent(params.url),
  ].join("/");

export const getUploadedImageUrl = (params: CloudinaryUploadUrlParams) =>
  [
    CLOUDINARY_ORIGIN,
    params.cloudName ?? CLOUDINARY.defaultCloudName,
    params.type ?? CloudinaryMedia.IMAGE,
    "upload",
    getParamsString(params),
    `${params.version}${CLOUDINARY.prefix ?? ""}${params.publicId}`,
  ].join("/");

type SrcSetFactory = {
  (fn: typeof getFetchedImageUrl): (
    params: CloudinaryFetchedImageSrcSetParams
  ) => string;
  (fn: typeof getUploadedImageUrl): (
    params: CloudinaryUploadSrcSetParams
  ) => string;
};
type SrcSetParams =
  | CloudinaryFetchedImageSrcSetParams
  | CloudinaryUploadSrcSetParams;

const srcSetFactory: SrcSetFactory = (
  fn: any //eslint-disable-line @typescript-eslint/no-explicit-any
) => ({ widths, ...rest }: SrcSetParams) =>
  widths
    .map(
      (width) =>
        `${fn({
          ...rest,
          width,
        })} ${width}w`
    )
    .join(",");

/** @deprecated Use `ResponsiveImage` instead */
export const getFetchedImageSrcSet = srcSetFactory(getFetchedImageUrl);

/** @deprecated Use `ResponsiveImage` instead */
export const getUploadedImageSrcSet = srcSetFactory(getUploadedImageUrl);
