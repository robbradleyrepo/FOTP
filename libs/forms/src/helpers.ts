export const removeEmptyFields = <T extends Record<keyof T, unknown>>(
  values: T
): Partial<T> =>
  Object.entries(values ?? {}).reduce(
    (accum, [key, value]) =>
      value === undefined || value === null
        ? accum
        : { ...accum, [key]: value },
    {}
  );
