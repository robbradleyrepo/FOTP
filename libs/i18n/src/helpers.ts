import { ParsedUrlQuery } from "querystring";

import { getConfig } from "./config";
import { Locale } from "./types";

const findLocaleByNextLocale = ({
  locale,
  defaultLocale,
}: {
  locale?: string;
  defaultLocale?: string;
}): Locale | undefined => {
  const langtag = locale ?? defaultLocale;
  return langtag ? findLocaleByLangtag(langtag) : undefined;
};

const findLocaleByLangtag = (langtag: string) => {
  const { supportedLocales } = getConfig();

  return supportedLocales.find(
    (locale) => locale.langtag.toLowerCase() === langtag.toLowerCase()
  );
};

// In most cases we'll want to use the locale value provided by Next's page
// or app context, but we'll still need to use this for the API as it doesn't
// have access to either of these.
const findLocaleByQuery = (query: ParsedUrlQuery) => {
  const { supportedLocales } = getConfig();

  return supportedLocales.find(
    ({ language, region }) =>
      language === query.language && region === query.region
  );
};

const getOpenGraphLocale = ({
  language,
  region,
}: Pick<Locale, "language" | "region">) =>
  `${language.toLowerCase()}_${region.toUpperCase()}`;

export {
  findLocaleByLangtag,
  findLocaleByNextLocale,
  findLocaleByQuery,
  getOpenGraphLocale,
};
