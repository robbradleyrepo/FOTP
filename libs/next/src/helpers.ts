import { ParsedUrlQuery, stringify } from "querystring";

enum PathType {
  CATCH_ALL = "CATCH_ALL",
  DYNAMIC = "DYNAMIC",
  OPTIONAL_CATCH_ALL = "OPTIONAL_CATCH_ALL",
}

export const getAsPath = (
  route: string,
  query?: ParsedUrlQuery,
  unmatched: "append" | "discard" = "append"
) => {
  // Make sure we haven't received a query string
  const [path, qs] = route.split("?");

  if (process.env.NODE_ENV !== "production" && qs) {
    throw new Error("The provided route must not include a query string");
  }

  const placeholders = new Map<string, { fragment: string; type: PathType }>();

  // Match all placeholders
  (path.match(/\/(\[[^/]+?\])(?=\/|$)/g) ?? []).forEach((match) => {
    const fragment = match.slice(1); // Remove leading backslash

    let key = fragment.slice(1, -1); // Remove square brackets
    let type = PathType.DYNAMIC;

    if (key.startsWith("...")) {
      key = key.slice(3);
      type = PathType.CATCH_ALL;
    }

    if (key.startsWith("[...") && key.endsWith("]")) {
      key = key.slice(4, -1);
      type = PathType.OPTIONAL_CATCH_ALL;
    }

    if (placeholders.has(key)) {
      throw new Error("Route contains duplicate param name");
    }

    placeholders.set(key, { fragment, type });
  });

  let asPath = path;
  const interpolables: ParsedUrlQuery = {
    ...query,
  };
  const unmatchedParams = new Set(Object.keys(interpolables));

  placeholders.forEach(({ fragment, type }, key) => {
    let value: string | string[] | undefined = interpolables[key];

    if (type === PathType.OPTIONAL_CATCH_ALL && !value) {
      value = [];
    }

    if (!value) {
      throw new Error("Missing query data for dynamic route");
    }

    if (Array.isArray(value) && type === PathType.DYNAMIC) {
      throw new Error("Arrays can only be used for catch-all routes");
    }

    // Encode and stringify values
    const parsedValue = Array.isArray(value)
      ? value.map(encodeURIComponent).join("/")
      : encodeURIComponent(value);

    asPath = asPath.replace(fragment, parsedValue).replace(/\/$/, "");

    unmatchedParams.delete(key);
  });

  if (unmatched === "append") {
    if (unmatchedParams.size > 0) {
      const query = [...unmatchedParams.keys()].reduce(
        (accum, key) => ({ ...accum, [key]: interpolables[key] }),
        {} as ParsedUrlQuery
      );
      asPath += `?${stringify(query)}`;
    }
  }

  return asPath;
};
