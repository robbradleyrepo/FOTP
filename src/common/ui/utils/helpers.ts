import { captureException } from "@sentry/nextjs";
import * as CSS from "csstype";
import facepaint, { DynamicStyleFunction } from "facepaint";
import { getContrast } from "polished";

import { evaluate, isObject } from "./common";
import { useTheme } from "./provider";
import {
  CSSValue,
  ResponsiveCSSObject,
  ResponsiveCSSObjectOrValue,
  ResponsiveCSSValue,
  Style,
  StyleFn,
  Theme,
} from "./types";

const getBreakpoints = ({ breakpoint }: Theme) => [
  breakpoint.sm,
  breakpoint.md,
  breakpoint.lg,
  breakpoint.xl,
];

export const getDynamicStyleFn = (theme: Theme): DynamicStyleFunction => {
  const breakpoints = getBreakpoints(theme);

  return facepaint(
    breakpoints.map((minWidth) => `@media(min-width:${minWidth}px)`)
  );
};

export const getMostContrasting = (reference: string, options: string[]) =>
  options
    .sort((a, b) => getContrast(reference, a) - getContrast(reference, b))
    .pop();

export const unresponsiveFactory = (
  properties: (keyof CSS.Properties)[] = []
) => {
  const unresponsiveProperties = new Set<string>(properties);

  const unresponsive = (style: ResponsiveCSSObject): ResponsiveCSSObject =>
    Object.entries(style).reduce((accum, [key, value]) => {
      let result: ResponsiveCSSObjectOrValue = value;

      if (Array.isArray(result)) {
        if (unresponsiveProperties.has(key)) {
          result = result[0];
        }
      } else if (isObject(result)) {
        result = unresponsive(result);
      }

      return { ...accum, [key]: result };
    }, {});

  return (style: Style): StyleFn => (t) => unresponsive(evaluate(style, t));
};

export const unresponsiveTypography = unresponsiveFactory([
  "fontFamily",
  "fontSize",
  "fontStyle",
  "fontWeight",
  "letterSpacing",
  "lineHeight",
]);

export const useContrastingBackground = (color: string) => {
  const theme = useTheme();

  return getMostContrasting(color, Object.values(theme.color.background));
};

export const useContrastingText = (color: string) => {
  const theme = useTheme();

  return getMostContrasting(color, [
    theme.color.text.light.base,
    theme.color.text.dark.base,
  ]);
};

export const getValueAtBreakpoint = (
  responsiveValues: ResponsiveCSSValue,
  index: number
) => {
  if (!Array.isArray(responsiveValues)) {
    return responsiveValues;
  }

  // Use the same mapping as `facepaint` for consistency with our styling
  // system: ignore `null` or `undefined` values, and fallback to the last
  // valid value
  for (let i = index; i >= 0; i -= 1) {
    const value = responsiveValues[i];

    if (value !== null && value !== undefined) {
      return value;
    }
  }

  return null;
};

interface Width {
  unit: "px" | "viewport";
  value: number;
}

const toFloat = (value: string) => {
  const result = parseFloat(value);

  if (isNaN(result)) {
    throw new Error(`Could not convert string "${value}" into float`);
  }

  return result;
};

// The `sizes` attribute only accepts whole pixels
const toPx = (value: number) => `${Math.round(value)}px`;

const parseWidth = (cssValue: CSSValue): Width | null => {
  if (typeof cssValue === "number") {
    return {
      unit: "px",
      value: cssValue,
    };
  } else if (typeof cssValue === "string") {
    if (cssValue.endsWith("vw")) {
      return {
        unit: "viewport",
        value: toFloat(cssValue) / 100,
      };
    } else if (cssValue.endsWith("px")) {
      return {
        unit: "px",
        value: toFloat(cssValue),
      };
    } else if (cssValue.match(/^\d+\D+$/)) {
      const error = new Error(
        `Invalid length unit used in "${cssValue}": expected "px" or "vw"`
      );

      if (process.env.NODE_ENV === "production") {
        captureException(error);
      } else {
        throw error;
      }
    }
  }

  return null;
};

export interface ResponsiveImageProperties {
  // Note that these are the values applied to the image in the layout, not the
  // image's intrinsic dimensions or max-width of its containing element.
  // For example, consider the following layout:
  //
  // <div css={s({ maxWidth: 960 })>
  //   <img css={s({ width: "50%" })} />
  // </div>
  //
  // The image occupies 50% of its parent element, so its
  // `ResponsiveImageProperties` would be `{ maxWidth: 480, width: "50vw" }`
  maxWidth?: ResponsiveCSSValue;
  width?: ResponsiveCSSValue;
}

export const useResponsiveImageSizes = () => {
  const theme = useTheme();

  const breakpoints = getBreakpoints(theme);

  return (properties: ResponsiveImageProperties) => {
    const sizes = [null, ...breakpoints].reduce<
      {
        breakpoint: number | null;
        size: string;
      }[]
    >((accum, breakpoint, index, arr) => {
      const nextBreakpoint = arr[index + 1];
      const maxWidth = parseWidth(
        getValueAtBreakpoint(properties.maxWidth, index)
      );
      const width = parseWidth(getValueAtBreakpoint(properties.width, index));

      let sizes = accum;

      let maxViewportWidth = 1; // Default to 100vw
      let maxPixelWidth: number | null = null;

      if (maxWidth?.unit === "px") {
        maxPixelWidth = maxWidth.value;
      } else if (maxWidth?.unit === "viewport") {
        maxViewportWidth = maxWidth.value;
      }

      if (
        width?.unit === "px" &&
        (maxPixelWidth === null || maxPixelWidth > width.value)
      ) {
        maxPixelWidth = width.value;
      } else if (width?.unit === "viewport" && maxViewportWidth > width.value) {
        maxViewportWidth = width.value;
      }

      // If the image has a maximum pixel value, let's check to see if it has
      // already taken effect, or if it takes effect within this breakpoint
      if (maxPixelWidth) {
        // The viewport width (in pixels) at which the maximum width is reached
        const viewportWidth = maxPixelWidth / maxViewportWidth;

        if (breakpoint && viewportWidth < breakpoint) {
          sizes = [
            ...sizes,
            {
              breakpoint,
              size: toPx(maxPixelWidth),
            },
          ];
        } else {
          sizes = [
            ...sizes,
            {
              breakpoint,
              size: `${maxViewportWidth * 100}vw`,
            },
          ];

          if (!nextBreakpoint || viewportWidth < nextBreakpoint) {
            sizes = [
              ...sizes,
              {
                breakpoint: viewportWidth,
                size: toPx(maxPixelWidth),
              },
            ];
          }
        }
      } else {
        sizes = [
          ...sizes,
          {
            breakpoint,
            size: `${maxViewportWidth * 100}vw`,
          },
        ];
      }

      return sizes;
    }, []);

    return sizes.reduce((accum, { breakpoint, size }, index, arr) => {
      const lastSize = arr[index - 1];

      let str = accum;

      if (lastSize?.size !== size) {
        str = breakpoint ? `(min-width:${breakpoint}px) ${size}, ${str}` : size;
      }

      return str;
    }, "");
  };
};
