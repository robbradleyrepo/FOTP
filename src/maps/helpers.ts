import { LatLng } from "@sss/prismic";

import { color } from "../ui/styles/variables";

const getOverlay = ([...markers]: LatLng[]) =>
  encodeURIComponent(
    markers
      .sort((a, b) => b.latitude - a.latitude) // Sort by latitude, so southern flags sit in front of northern ones
      .map(
        ({ latitude, longitude }) =>
          `pin-s+${color.background.dark.replace(
            "#",
            ""
          )}(${longitude},${latitude})`
      )
      .join(",")
  )
    // Encode parentheses otherwise it will break Styled Components when used
    // as a responsive background image
    .replace(/\(/g, "%28")
    .replace(/\)/g, "%29");

export interface GetMapUrlArguments extends LatLng {
  height: number;
  markers: LatLng[];
  styleId?: string;
  username?: string;
  width: number;
  zoom: number;
}

// See https://docs.mapbox.com/api/maps/#static
export const getMapUrl = ({
  height,
  latitude,
  longitude,
  markers,
  styleId = process.env.MAPBOX_STYLE_ID,
  username = process.env.MAPBOX_USERNAME,
  width,
  zoom,
}: GetMapUrlArguments) =>
  `https://api.mapbox.com/styles/v1/${username}/${styleId}${
    process.env.NODE_ENV === "production" ? "" : "/draft"
  }/static/${getOverlay(
    markers
  )}/${longitude},${latitude},${zoom}/${width}x${height}@2x?access_token=${
    process.env.MAPBOX_KEY
  }`;
