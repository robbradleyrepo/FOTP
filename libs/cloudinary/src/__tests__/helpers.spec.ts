import { getFetchedImageUrl, getUploadedImageUrl } from "../helpers";
import {
  CloudinaryFetchedImageUrlParams,
  CloudinaryFlag,
  CloudinaryMedia,
  CloudinaryUploadUrlParams,
} from "../types";

jest.mock("../config", () => ({
  CLOUDINARY_ORIGIN: "https://res.cloudinary.com",
  __esModule: true,
  default: { defaultCloudName: "cumulus", prefix: "/alto" },
}));

describe("getFetchedImageUrl", () => {
  it("should generate a URL based on the provided parameters", () => {
    expect(
      getFetchedImageUrl({
        cloudName: "stratus",
        quality: 123,
        type: CloudinaryMedia.VIDEO,
        url: "https://www.fillmurray.com/720/720",
        width: 456,
      })
    ).toMatchSnapshot();
  });

  it("should use the `defaultCloudName` if no `cloudName` is provided", () => {
    expect(
      getFetchedImageUrl({
        quality: 123,
        type: CloudinaryMedia.IMAGE,
        url: "https://www.fillmurray.com/720/720",
        width: 456,
      })
    ).toMatchSnapshot();
  });

  it("should use `auto` if no `quality` is provided", () => {
    expect(
      getFetchedImageUrl({
        cloudName: "stratus",
        type: CloudinaryMedia.IMAGE,
        url: "https://www.fillmurray.com/720/720",
        width: 456,
      })
    ).toMatchSnapshot();
  });

  it("should use `image` if no `type` is provided", () => {
    expect(
      getFetchedImageUrl({
        cloudName: "stratus",
        quality: 123,
        url: "https://www.fillmurray.com/720/720",
        width: 456,
      })
    ).toMatchSnapshot();
  });

  it("should use `auto` if no `width` is provided", () => {
    expect(
      getFetchedImageUrl({
        cloudName: "stratus",
        quality: 123,
        type: CloudinaryMedia.IMAGE,
        url: "https://www.fillmurray.com/720/720",
      })
    ).toMatchSnapshot();
  });

  it("should include flags in a consistent order", () => {
    const params: CloudinaryFetchedImageUrlParams = {
      cloudName: "stratus",
      quality: 123,
      type: CloudinaryMedia.IMAGE,
      url: "https://www.fillmurray.com/720/720",
      width: 456,
    };

    const x = getFetchedImageUrl({
      ...params,
      flags: [
        CloudinaryFlag.ANY_FORMAT,
        CloudinaryFlag.CLIP,
        CloudinaryFlag.LOSSY,
      ],
    });
    const y = getFetchedImageUrl({
      ...params,
      flags: [
        CloudinaryFlag.LOSSY,
        CloudinaryFlag.ANY_FORMAT,
        CloudinaryFlag.CLIP,
      ],
    });

    expect(x).toBe(y);
    expect(x).toMatchSnapshot();
  });

  it("should handle an empty array of flags", () => {
    expect(
      getFetchedImageUrl({
        cloudName: "stratus",
        flags: [],
        quality: 123,
        type: CloudinaryMedia.IMAGE,
        url: "https://www.fillmurray.com/720/720",
        width: 456,
      })
    ).toMatchSnapshot();
  });
});

describe("getUploadedImageUrl", () => {
  it("should generate a URL based on the provided parameters", () => {
    expect(
      getUploadedImageUrl({
        cloudName: "stratus",
        publicId: "/foo",
        quality: 123,
        type: CloudinaryMedia.VIDEO,
        version: "v1",
        width: 456,
      })
    ).toMatchSnapshot();
  });

  it("should use the `defaultCloudName` if no `cloudName` is provided", () => {
    expect(
      getUploadedImageUrl({
        publicId: "/foo",
        quality: 123,
        type: CloudinaryMedia.IMAGE,
        version: "v1",
        width: 456,
      })
    ).toMatchSnapshot();
  });

  it("should use `auto` if no `quality` is provided", () => {
    expect(
      getUploadedImageUrl({
        cloudName: "stratus",
        publicId: "/foo",
        type: CloudinaryMedia.IMAGE,
        version: "v1",
        width: 456,
      })
    ).toMatchSnapshot();
  });

  it("should use `image` if no `type` is provided", () => {
    expect(
      getUploadedImageUrl({
        cloudName: "stratus",
        publicId: "/foo",
        quality: 123,
        version: "v1",
        width: 456,
      })
    ).toMatchSnapshot();
  });

  it("should use `auto` if no `width` is provided", () => {
    expect(
      getUploadedImageUrl({
        cloudName: "stratus",
        publicId: "/foo",
        quality: 123,
        type: CloudinaryMedia.IMAGE,
        version: "v1",
      })
    ).toMatchSnapshot();
  });

  it("should include flags in a consistent order", () => {
    const params: CloudinaryUploadUrlParams = {
      cloudName: "stratus",
      publicId: "/foo",
      quality: 123,
      type: CloudinaryMedia.IMAGE,
      version: "v1",
      width: 456,
    };

    const x = getUploadedImageUrl({
      ...params,
      flags: [
        CloudinaryFlag.ANY_FORMAT,
        CloudinaryFlag.CLIP,
        CloudinaryFlag.LOSSY,
      ],
    });
    const y = getUploadedImageUrl({
      ...params,
      flags: [
        CloudinaryFlag.LOSSY,
        CloudinaryFlag.ANY_FORMAT,
        CloudinaryFlag.CLIP,
      ],
    });

    expect(x).toBe(y);
    expect(x).toMatchSnapshot();
  });

  it("should handle an empty array of flags", () => {
    expect(
      getUploadedImageUrl({
        cloudName: "stratus",
        flags: [],
        publicId: "/foo",
        quality: 123,
        type: CloudinaryMedia.IMAGE,
        version: "v1",
        width: 456,
      })
    ).toMatchSnapshot();
  });
});
