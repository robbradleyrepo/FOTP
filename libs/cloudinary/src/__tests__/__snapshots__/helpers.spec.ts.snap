// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFetchedImageUrl should generate a URL based on the provided parameters 1`] = `"https://res.cloudinary.com/stratus/video/fetch/f_auto,q_123,w_456/https%3A%2F%2Fwww.fillmurray.com%2F720%2F720"`;

exports[`getFetchedImageUrl should handle an empty array of flags 1`] = `"https://res.cloudinary.com/stratus/image/fetch/f_auto,q_123,w_456/https%3A%2F%2Fwww.fillmurray.com%2F720%2F720"`;

exports[`getFetchedImageUrl should include flags in a consistent order 1`] = `"https://res.cloudinary.com/stratus/image/fetch/f_auto,fl_any_format,fl_clip,fl_lossy,q_123,w_456/https%3A%2F%2Fwww.fillmurray.com%2F720%2F720"`;

exports[`getFetchedImageUrl should use \`auto\` if no \`quality\` is provided 1`] = `"https://res.cloudinary.com/stratus/image/fetch/f_auto,q_auto,w_456/https%3A%2F%2Fwww.fillmurray.com%2F720%2F720"`;

exports[`getFetchedImageUrl should use \`auto\` if no \`width\` is provided 1`] = `"https://res.cloudinary.com/stratus/image/fetch/f_auto,q_123,w_auto/https%3A%2F%2Fwww.fillmurray.com%2F720%2F720"`;

exports[`getFetchedImageUrl should use \`image\` if no \`type\` is provided 1`] = `"https://res.cloudinary.com/stratus/image/fetch/f_auto,q_123,w_456/https%3A%2F%2Fwww.fillmurray.com%2F720%2F720"`;

exports[`getFetchedImageUrl should use the \`defaultCloudName\` if no \`cloudName\` is provided 1`] = `"https://res.cloudinary.com/cumulus/image/fetch/f_auto,q_123,w_456/https%3A%2F%2Fwww.fillmurray.com%2F720%2F720"`;

exports[`getUploadedImageUrl should generate a URL based on the provided parameters 1`] = `"https://res.cloudinary.com/stratus/video/upload/f_auto,q_123,w_456/v1/alto/foo"`;

exports[`getUploadedImageUrl should handle an empty array of flags 1`] = `"https://res.cloudinary.com/stratus/image/upload/f_auto,q_123,w_456/v1/alto/foo"`;

exports[`getUploadedImageUrl should include flags in a consistent order 1`] = `"https://res.cloudinary.com/stratus/image/upload/f_auto,fl_any_format,fl_clip,fl_lossy,q_123,w_456/v1/alto/foo"`;

exports[`getUploadedImageUrl should use \`auto\` if no \`quality\` is provided 1`] = `"https://res.cloudinary.com/stratus/image/upload/f_auto,q_auto,w_456/v1/alto/foo"`;

exports[`getUploadedImageUrl should use \`auto\` if no \`width\` is provided 1`] = `"https://res.cloudinary.com/stratus/image/upload/f_auto,q_123,w_auto/v1/alto/foo"`;

exports[`getUploadedImageUrl should use \`image\` if no \`type\` is provided 1`] = `"https://res.cloudinary.com/stratus/image/upload/f_auto,q_123,w_456/v1/alto/foo"`;

exports[`getUploadedImageUrl should use the \`defaultCloudName\` if no \`cloudName\` is provided 1`] = `"https://res.cloudinary.com/cumulus/image/upload/f_auto,q_123,w_456/v1/alto/foo"`;
